### PR TITLE
Create simple matcher utility function for dataToQuery and migrate existing apps

### PR DESCRIPTION
**Pull request by OrkoHunter**
Closes https://github.com/felvin-search/instant-apps/issues/84

Also -
* Added some logs for user next steps in `yarn create-app` script.
* Disabled noEmit so that dist folder exists when publishing 
* Converted shell scrip to node script to enable multi OS support as well simpler scripting syntax in node.

### CODE REVIEW

File: utils/create_new_app.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env node
```
Comment (hargup) [inline utils/create_new_app.js:1]: +1 to converting this file to JS. Its best to stick with one language/ecosystem, helps with reusing utilities.

File: src/lib/utilityApis.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      } else {
+        if (query.toLowerCase().includes(triggerQuery.toLowerCase())) {
+          return { query };
+        }
+      }
+    }
+
+    return;
+  };
+}
```
Comment (hargup) [inline src/lib/utilityApis.ts:63]: This is nice!
The options `caseSensitive` and `subStringMatch` can surely be useful.

**Review by hargup — APPROVED** [review]
@OrkoHunter Looks great, here's a taco for you 🌮

**Comment by hargup** [comment]
The next step will be adding this to the documentation, though the PR doesn't need to wait for it.

**Review by sahil-shubham — APPROVED** [review]
Looks good to me. :shipit:

**Pull request merged**